packages/react-components/react-color-picker-preview/stories/src/ColorPicker/ColorPickerDefault.stories.tsx

```diff
@@ -36,23 +45,51 @@ const useStyles = makeStyles({
   input: {
     width: '80px',
   },
+  spinButton: {
+    width: '50px',
+  },
 });
 
 const HEX_COLOR_REGEX = /^#?([0-9A-Fa-f]{0,6})$/;
+const NUMBER_REGEX = /^\d+$/;
 const DEFAULT_COLOR_HSV = tinycolor('#2be700').toHsv();
 
+type RgbKey = 'r' | 'g' | 'b';
+
 export const Default = () => {
   const hexId = useId('hex-input');
 
   const styles = useStyles();
   const [color, setColor] = React.useState(DEFAULT_COLOR_HSV);
   const [hex, setHex] = React.useState(tinycolor(color).toHexString());
+  const [rgb, setRgb] = React.useState(tinycolor(color).toRgb());
+  const [red, setRed] = React.useState(rgb.r);
```

Comment:
No need to store state for `red/green/blue` since it's already in `rgb`.



```tsx
const parseRGBChannelValue = (value: string, min = 0, max = 255) => {
   if (/^\d+$/.test(data.value)) {
     return null;
   }
    
   const parsedValue = parseInt(value, 10);

   if (isNaN(parsedValue)) {
     return null;
   }
   
   return clamp(parsedValue, min, max);
}

const onRgbChange: InputRgbFieldProps<'onChange'> = (event, data) => {
    const value = parseRGBChannelValue(data.value);
    
    // new value is not valid, we shouldn't update state
    if (value === null) {
      return;
    }
    
    // get color key value from the element attribute 
    const colorKey = event.target.name as RgbKey;

    // create new color
    const newColor = tinycolor({ ...rgb, [colorKey]: value });
    
    // update the new color in the component state
    if (newColor.isValid) {
      setColor(newColor.toHsv());
      setHex(newColor.toHex());
      setRgb(newColor.toRgb());
    }
}


// and then

<InputRgbField label="Red" value={rgb.r} name="r" onChange={onRgbChange} />
<InputRgbField label="Green" value={rgb.g} name="g" onChange={onRgbChange} />
<InputRgbField label="Blue" value={rgb.b} name="b" onChange={onRgbChange} />

```

Also, you can remove `handleRgbKeyPress` as it's not needed anymore, because we set to state only valid values in onChange

Reply:
`parseRGBChannelValue` is not working. Probably, because SpinButton has `value` as number and `displayValue` as string. But I'll try to find a workaround to make it work

Reply:
is there a way to get the actual input value (string) in the onChange handler, 'event.target.value' possibly?

Reply:
I've tried it at the beginning. With `event.target.value` SpinButton stops working at all